code/modules/cargo/randomstock.dm

```diff
@@ -0,0 +1,1880 @@
+	"suspension" = 2,
+	"animal" = 2.5,
+	"cablelayer" = 1,
+	"floodlight" = 3,
+	"floorlayer" = 2,
+	"heater" = 1.3,
+	"dispenser" = 2.5,
+	"jukebox" = 1.2,
+	"pipemachine" = 1.7,
+	"nothing" = 0)
+
+
+/proc/spawn_cargo_stock()
+	var/start_time = world.timeofday
+	new /datum/cargospawner()
+	admin_notice("<span class='danger'>Cargo Stock generation completed in [round(0.1*(world.timeofday-start_time),0.1)] seconds.</span>", R_DEBUG)
+
+/datum/cargospawner
+	var/list/containers = list()
+	var/list/tables = list()
+	var/list/full_containers = list()//Used to hold references to crates we filled up
+	var/area/warehouse
+
+	var/list/infest_mobs_minor = list(
+		"/mob/living/simple_animal/mouse" = 1,
+		"/mob/living/simple_animal/lizard" = 0.5,
+		"/mob/living/simple_animal/yithian" = 0.7,
+		"/mob/living/simple_animal/tindalos" = 0.6,
+		"/mob/living/bot/secbot" = 0.1)
+
+	var/list/infest_mobs_moderate = list(
+		"/mob/living/simple_animal/bee" = 1,
+		"/mob/living/simple_animal/hostile/diyaab" = 1,
+		"/mob/living/simple_animal/hostile/viscerator" = 1,
+		"/mob/living/simple_animal/hostile/scarybat" = 1)
+
+	var/list/infest_mobs_severe = list(
+		"/mob/living/simple_animal/hostile/giant_spider/hunter" = 1,
+		"/mob/living/simple_animal/hostile/shantak" = 0.7,
+		"/mob/living/simple_animal/hostile/bear" = 0.5,
+		"/mob/living/simple_animal/hostile/carp" = 1.5,
+		"cratey" = 1
+	)
+
+/datum/cargospawner/New()
+
+	//First lets get the reference to our warehouse
+	for(var/areapath in typesof(/area/quartermaster/storage))
+		warehouse = locate(areapath)
+
+	if (!warehouse)
+		admin_notice("<span class='danger'>ERROR: Cargo spawner failed to locate warehouse. Terminating.</span>", R_DEBUG)
+		qdel(src)
+		return
+
+	for (var/obj/structure/closet/crate/C in warehouse)
+		containers |= C
+
+	for (var/obj/structure/table/B in warehouse)
+		tables |= B
+
+	//First, we spawn the larger items
+	//Large objects are spawned on preset locations around cargo
+	//These locations are designated by large stock marker objects, which are manually mapped in
+	for (var/obj/effect/large_stock_marker/LSM in world)
+		if (prob(STOCK_LARGE_PROB))
+			spawn_stock(pickweight(random_stock_large), get_turf(LSM))
+			qdel(LSM)
+
+
+	//Now we spawn the smaller items
+	//These are spawned inside the cargo warehouse, in crates and on tables.
+	var/spawns = TOTAL_STOCK
+	while (spawns > 0)
+		var/atom/spawnloc = get_location()
+		var/spawntype = get_spawntype()
+		spawn_stock(spawntype, spawnloc, src)
+		spawns--
+
+	handle_infestation()
+	shuffle_items()
+
+/datum/cargospawner/proc/get_location()
+	var/cratespawn = 0
+	var/obj/structure/closet/crate/emptiest
+	if (prob(70))//We'll usually put items in crates
+		var/minweight = 99999999999//We will distribute items somewhat evenly among crates
+		//by selecting the least-filled one for each spawn
+
+		for (var/obj/structure/closet/crate/C in containers)
+			if (C.stored_weight() < minweight && C.stored_weight() <= C.storage_capacity)
```

Comment:
`C.stored_weight() <= C.storage_capacity`
should be < and not <=
Why spawn in a new item if the stored weight already equals the storage capacity